sdk/python/feast/job.py

```diff
@@ -128,14 +132,48 @@ def result(self, timeout_sec: int = int(defaults[CONFIG_TIMEOUT_KEY])):
         Returns:
             Iterable of Avro rows.
         """
-        uris = self.get_avro_files(timeout_sec)
+        # list of parseResult
+        uris = self.get_files(timeout_sec)
         for file_uri in uris:
             file_obj = get_staging_client(file_uri.scheme).download_file(file_uri)
             file_obj.seek(0)
-            avro_reader = fastavro.reader(file_obj)
+            for row in self.result_object_reader(file_obj, file_uri):
+                yield row
 
+    def result_object_reader(self, file_obj: BufferedRandom, file_uri: ParseResult):
+        """
+        iterable of result object rows
+        :param file_obj: BufferedRandom: result object
+        :return: Iterable of Avro or csv rows.
+        """
+        if self.job_proto.data_format == DATA_FORMAT_AVRO:
+            avro_reader = fastavro.reader(file_obj)
             for record in avro_reader:
                 yield record
+        elif self.job_proto.data_format == DATA_FORMAT_CSV:
+            # check file format only support csv and gz compression files
+            ext = file_uri.path.lstrip('/').split('.')[-1]
+            if ext == 'csv':
+                decode_obj = file_obj.read().decode("utf-8")
+                csv_reader = csv.DictReader(decode_obj.splitlines(), delimiter=',')
+                for record in csv_reader:
+                    yield record
+            elif ext == 'gz':
+                with gzip.open(file_obj, mode='rt') as f:
+                    csv_reader = csv.DictReader(f, delimiter=',')
+                    for record in csv_reader:
+                        yield record
+            else:
+                raise Exception(
+                    "Feast only supports CSV data format with .csv or .csv.gz extension for now. Please check "
+                    "your Feast Serving deployment."
+                )
+        else:
```

Comment:
is this necessary?

Reply:
I think we need to add an exception case to check when the file is not in any of the above file formats